src/ShopBundle/objects/db/TShopShippingGroup.class.php

```diff
@@ -423,18 +423,21 @@ public function GetVat()
         if (is_null($oVat)) {
             $oVat = null;
             $oShopConf = TdbShop::GetInstance();
-            if (!$oShopConf->fieldShippingVatDependsOnBasketContents) {
-                $oVat = $this->GetFieldShopVat();
-                if (is_null($oVat)) {
-                    $oShopConf = TdbShop::GetInstance();
-                    $oVat = $oShopConf->GetVat();
-                }
-                $this->SetInternalCache('ovat', $oVat);
-            } else {
-                // use max vat in basket... if we have contents
+            if (true === $oShopConf->fieldShippingVatDependsOnBasketContents) {
                 $oBasket = TShopBasket::GetInstance();
                 $oVat = $oBasket->GetLargestVATObject();
+                if(null !== $oVat) {
```

Comment:
Falls es === null ist, wird der Cache aber nie gesetzt?

Reply:
Doch dann wird ja drunter der Vat aus der Shipping group geholt und dann in den cache geschrieben